packages/blade/src/components/Checkbox/CheckboxInput.web.tsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import type { CSSObject } from 'styled-components';
+import styled from 'styled-components';
+import getIn from '../../utils/getIn';
+import makeMotionTime from '../../utils/makeMotionTime';
+import type { Theme } from '../BladeProvider';
+import { screenReaderStyles } from '../VisuallyHidden';
+import type { CheckboxIconProps } from './CheckboxIcon';
+import type { InputProps } from './useCheckbox';
+
+type HoverProps = Pick<CheckboxIconProps, 'isNegative' | 'isDisabled'> & {
+  isChecked: boolean;
+};
+
+const variants = {
+  default: {
+    background: {
+      checked: 'colors.brand.primary.600',
+      unchecked: 'colors.brand.gray.400',
+    },
+  },
+};
+
+const getHoverStyles = ({
+  theme,
+  isDisabled,
+  isNegative,
+  isChecked,
+}: { theme: Theme } & HoverProps): CSSObject => {
+  if (isDisabled || isNegative) return {};
+
+  const checked = isChecked ? 'checked' : 'unchecked';
+  const backgroundColor = variants.default.background[checked];
+
+  return {
+    backgroundColor: getIn(theme, backgroundColor),
+    transitionTimingFunction: theme.motion.easing.standard.effective as string,
+    transitionDuration: makeMotionTime(theme.motion.duration['2xquick']),
+  };
+};
+
+const StyledInput = styled.input<HoverProps>(({ theme, isChecked, isDisabled, isNegative }) => ({
+  ...screenReaderStyles,
+  '&:focus + div': {
+    outline: '1px solid white',
+    boxShadow: `0px 0px 0px 4px ${theme.colors.brand.primary[400]}`,
+  },
+  '&:hover + div, &:focus + div': {
+    ...getHoverStyles({ theme, isChecked, isDisabled, isNegative }),
+  },
```

Comment:
`&:focus + div` will show that focus ring outside of the `CheckboxIcon` 

Initially I did it with JS with a custom useFocusWithin hook but then released since reactnative doesn't need focus ring it's okay to go with just pure CSS.

Reply:
CSS :godmode:

Reply:
:godmode: